jyulb/internal/isothermal/jyu_engine.py

```diff
@@ -313,51 +235,31 @@ def iter_lattices(self, names=None):
             names of specific lattices to be iterated over. If names is not
             given, then all lattices will be iterated over.
 
-        Yields
-        -------
-        ABCLattice
-        """
-        yield self._lat
-
-    def iter_meshes(self, names=None):
-        """Iterate over a subset or all of the meshes.
-
-        Parameters
-        ----------
-        names : sequence of str, optional
-            names of specific meshes to be iterated over. If names is not
-            given, then all meshes will be iterated over.
-
-        Yields
+        Returns
         -------
-        ABCMesh
+        A generator of ABCLattice objects
 
         Raises
         ------
-        NotImplementedError
-           always.
+        ValueError
+            if any one of the names is not in SD
         """
-        message = 'JYUEngine does not handle meshes'
-        raise NotImplementedError(message)
-
-    def iter_particles(self, names=None):
-        """Iterate over a subset or all of the particle containers.
+        self._update_dataset_names()
+        if names is None:
+            for name in self.SD.keys():
+                yield self.SD[name]
+        else:
+            for name in names:
+                if name not in self.SD.keys():
+                    message = 'State data does not contain requested item'
+                    raise ValueError(message)
+                yield self.SD[name]
+
+    def _update_dataset_names(self):
```

Comment:
> This would mean, among other things, that the SDs are not directly maintained by the engines themselves. 

I would still think that the SD would be maintained by the engines.  So the design of the engine might change but at the end of the day, we still want these proxy datasets as we want the user to be able to query/update parts of the SD between "run" calls of the engine  Or at least that is my understanding.  Maybe @ahashibon can clarify more.  

>A question related to the current discussion then is, how the "single lattice dataset" can be enforced with jyu-lb engines? 

I think either by throwing an exception when either (i) the "run" is called that there is not a single lattice or (ii) throw an exception when user tries to add a second dataset (which the current implemented solution).